sourcegraph/sourcegraph
# Code Insights: Refactor code insights types

## Context

This PR removes all old legacy Code Insights related models in TS interfaces that we introduced some time ago for the Setting Based API that already is outdated. 

- Removes wide interfaces from common API interfaces that we had for the setting cascade operations
- Enables caching over backend insights data (insights that we have in the case now render immediately between dashboards)
- Removes old approach for the different running context at some API methods (we had them for legacy functionality where we render insights in different places - home page, repository page, ...) Currently, this is outdated and had been removed by @unclejustin 
- Simplifies add/remove insights view, prior to this PR we used fetching all insights with all metadata fields. In this PR we fetch only field that we have to have for this particular view (insight title and ID fields)

## Test plan

- Make sure that all test passes
- Make sure that all happy path of the code insights are working properly 
- Make sure that apollo cache doesn't break anything around showing insights with outdated data
